### ISSUE TITLE
VLM captioning loss collapses to 0: labels not shifted for next-token

### ISSUE THREAD

**Issue by amazloumi** [KempnerInstitute/KempnerForge#154]
## What happened
`_tokenize_and_mask` aligns labels to the input position (`labels[i] == input_ids[i]`), but the loss doesn't shift and JD/MoT/MoMa return logits over the text positions (`output_slice=slice(n, None)`). So each logit is scored against the current, already-visible token — the model copies it and loss collapses to `0.0000` after warmup. Text pretraining shifts labels (`data/dataset.py`); the VLM path doesn't. Affects all captioning arches.

## Reproduction
```bash
uv run torchrun --nproc_per_node=4 scripts/train.py \
    examples/qwen3_vlm/configs/vlm_qwen3_0.6b_joint_decoder.toml \
    --train.max_steps=600 --scheduler.warmup_steps=100
```
loss → 0.0000 by ~step 400 while data/text_tokens_trained stays positive.

## Error / traceback

No traceback (silent). loss 8.6 (s40) → 0.0015 (s380) → 0.0000 (s980), grad_norm → ~0.

## Environment
- GPU/arch-independent; reproduces on multimodal-video and main.
